ams/CleanupCommand.cs

```diff
@@ -0,0 +1,226 @@
+﻿using AMSMigrate.Ams;
+using AMSMigrate.Contracts;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.Media;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+using Spectre.Console;
+
+namespace AMSMigrate.ams
+{
+    internal class CleanupCommand : BaseMigrator
+    {
+        private readonly ILogger _logger;
+        private readonly CleanupOptions _options;
+        private readonly IMigrationTracker<BlobContainerClient, AssetMigrationResult> _tracker;
+
+        public CleanupCommand(GlobalOptions globalOptions,
+            CleanupOptions cleanupOptions,
+            IAnsiConsole console,
+            TokenCredential credential,
+            IMigrationTracker<BlobContainerClient, AssetMigrationResult> tracker,
+            ILogger<CleanupCommand> logger)
+            : base(globalOptions, console, credential)
+        {
+            _options = cleanupOptions;
+            _logger = logger;
+            _tracker = tracker;
+        }
+        public override async Task MigrateAsync(CancellationToken cancellationToken)
+        {
+            var account = await GetMediaAccountAsync(_options.AccountName, cancellationToken);
+            _logger.LogInformation("Begin cleaning up on account: {name}", account.Data.Name);
+            Dictionary<string, bool> stats = new Dictionary<string, bool>();
+
+            var totalAssets = await QueryMetricAsync(
+                account.Id.ToString(),
+                "AssetCount",
+                cancellationToken: cancellationToken);
+
+            _logger.LogInformation("The total asset count of the media account is {count}.", totalAssets);
+            AsyncPageable<MediaAssetResource> assets;
+            if (!_options.IsCleanUpAccount)
+            {
+                var resourceFilter = GetAssetResourceFilter(_options.ResourceFilter, null, null);
+
+                var orderBy = "properties/created";
+                assets = account.GetMediaAssets()
+                    .GetAllAsync(resourceFilter, orderby: orderBy, cancellationToken: cancellationToken);
+            }
+            else
+            {
+                assets = account.GetMediaAssets()
+                   .GetAllAsync();
+            }
+            List<MediaAssetResource>? assetList = await assets.ToListAsync(cancellationToken);
+
+            foreach (var asset in assetList)
+            {
+                var result = await CleanUpAssetAsync(account, asset, cancellationToken);
+                stats.Add(asset.Data.Name, result);
+            }
+            WriteSummary(stats, false);
+            if (_options.IsCleanUpAccount)
+            {
+                Console.Write($"Do you want to delete the account '{account.Data.Name}'? (y/n): ");
+                string? userResponse = Console.ReadLine();
```

Comment:
Nit:  Can this be moved to the beginning when the code starts to handle the cleanup for the account? when customers response with No, there is no need to do any cleanup work, the tool would return immediately.